### PR TITLE
This makes all default initialization explicit 

### PR DESCRIPTION
**Pull request by lemire**
(which takes care of  good chunk of -Weff++ warnings).

### CODE REVIEW

**Comment by lemire** [comment]
@jkeiser  You may want to have a look. In my tests, this satisfies -Weffc++ in the sense that it no longer requires changes to the constructor. I think this is clean. I believe that it is the recommended way to do things in C++11.

File: include/simdjson/document.h [hunk 1]
```diff
@@ -373,13 +373,13 @@ class document {
   bool dump_raw_tape(std::ostream &os) const noexcept;
 
   /** @private Structural values. */
-  std::unique_ptr<uint64_t[]> tape;
+  std::unique_ptr<uint64_t[]> tape = {};
```
Comment (jkeiser) [inline include/simdjson/document.h:376]: Why are some of these `= {}` and and some of them `{}`?

**Comment by lemire** [comment]
Note that the failed test is a performance test, but I don't think it is possible for this PR to slow things down. I will benchmark to be sure, though.

File: include/simdjson/document.h (hunk 1 above)
Comment (lemire) [reply include/simdjson/document.h]: It is semantically equivalent but let us standardize.

**Comment by lemire** [comment]
Incredibly, there is a performance regression!!!

**Comment by lemire** [comment]
F*ck.

**Comment by jkeiser** [comment]
Oh God, I think I remember this happening to me too when I was futzing with initialization. If you tell C++ to initialize a member when the struct is initialized, and it enregisters the struct, maybe it's forced (or feels forced) to instantiate the registers early?

**Comment by lemire** [comment]
> Oh God, I think I remember this happening to me too when I was futzing with initialization. If you tell C++ to initialize a member when the struct is initialized, and it enregisters the struct, maybe it's forced (or feels forced) to instantiate the registers early?

I don't know. Locally, I was able to "fix" the performance regression by building my own default constructor... Let us see if it turns green. If it does, I will merge.

**Comment by lemire** [comment]
This is on hold. Maybe this PR will be dropped.

**Comment by lemire** [comment]
Retiring in favor of https://github.com/simdjson/simdjson/pull/764

**Pull request closed without merging**